src/groupeddataframe/complextransforms.jl

```diff
@@ -4,6 +4,9 @@ _nrow(x::NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}) =
 _ncol(df::AbstractDataFrame) = ncol(df)
 _ncol(x::Union{NamedTuple, DataFrameRow}) = length(x)
 
+_getnames(x::AbstractDataFrame) = _names(x)
+_getnames(x::Any) = throw(ArgumentError("names not defined for argument of type $(typeof(x))"))
```

Comment:
When can this happen?

Reply:
It should not happen. But I thought to add it to avoid `MethodError` but rather get an informative message. It will try to come up with a case when it is actually triggered :) (hopefully I will not find such).

Reply:
I have another question - should I group the `_getnames` definitions in one place?

The issue is that they are now spread around the code-base as they are defined where they were needed. The issue is that they refer to types that are introduced in different parts of the code base (a pain of not having DataFrames.jl having the code layout that we first define all types and then use them).

Reply:
Ah yeah if these overload the same function it would be clearer to have them in the same place.

Reply:
OK - I moved them, and I removed the `::Any` case as I could not think of the scenario when it would be triggered.